src/models/room.ts

```diff
@@ -914,37 +914,79 @@ export class Room extends ReadReceipt<RoomEmittedEvents, RoomEventHandlerMap> {
         return this.myUserId;
     }
 
+    /**
+     * Gets the "functional members" in this room.
+     *
+     * Returns the list of userIDs from the `io.element.functional_members` event. Does not consider the
+     * current membership states of those users.
+     *
+     * @see https://github.com/element-hq/element-meta/blob/develop/spec/functional_members.md.
+     */
+    private getFunctionalMembers(): string[] {
+        const mFunctionalMembers = this.currentState.getStateEvents(UNSTABLE_ELEMENT_FUNCTIONAL_USERS.name, "");
+        if (Array.isArray(mFunctionalMembers?.getContent().service_members)) {
+            return mFunctionalMembers!.getContent().service_members;
+        }
+        return [];
+    }
+
     public getAvatarFallbackMember(): RoomMember | undefined {
-        const memberCount = this.getInvitedAndJoinedMemberCount();
-        if (memberCount > 2) {
+        const functionalMembers = this.getFunctionalMembers();
+        // Optimizing for performance, we iterate over all members outermost, as is it the longest of the 3 lists we are comparing.
+        const nonFunctionalMemberCount = this.getMembers()!.reduce((count, m) => {
```

Comment:
I feel like a `for` loop would be easier to grok than `reduce` (functional programming is all very well, but it's not really JavaScript's thing, and it is often harder to read). YMMV though, no strong opinion.